### PR TITLE
Support CDN

### PR DESCRIPTION
**Pull request by tung491**

### CODE REVIEW

File: cdn.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	defer resp.Body.Close()
+	var data *DomainsResp
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (c *cdnService) Get(ctx context.Context, domainID string) (*ExtendedDomain, error) {
+	req, err := c.client.NewRequest(ctx, http.MethodGet, cdnName, strings.Join([]string{domainPath, domainID}, "/"), nil)
```
Comment (greatbn) [inline cdn.go:166]: Can we use itemPath function in here ?

File: cdn.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+type DomainsResp struct {
+	Domains []Domain `json:"results"`
+	Next    string   `json:"next"`
+	Prev    string   `json:"prev"`
+	Pages   int      `json:"pages"`
+	Total   int      `json:"total"`
+}
+
+type BandwidthPrice struct {
```
Comment (greatbn) [inline cdn.go:88]: what is this?

File: cdn.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+	ID        int    `json:"id"`
+	Email     string `json:"email"`
+	TenantID  string `json:"tenant_id"`
+	CreatedAt string `json:"created_at"`
+	Name      string `json:"name"`
+	StartedAt string `json:"started_at"`
+	Status    string `json:"status"`
+}
+
+type UsageUser struct {
```
Comment (greatbn) [inline cdn.go:58]: please check unused variable, struct?

**Review by greatbn — CHANGES_REQUESTED** [review]

File: cdn.go [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+	DomainID       string `json:"domain_id"`
+	Domain         string `json:"domain"`
+	Type           string `json:"type"`
+}
+
+type DomainsResp struct {
+	Domains []Domain `json:"results"`
+	Next    string   `json:"next"`
+	Prev    string   `json:"prev"`
+	Pages   int      `json:"pages"`
```
Comment (greatbn) [inline cdn.go:84]: Because these responses are paginate, Do you mind change the function to work with multiple pages?

File: client.go [hunk 5]
```diff
@@ -40,8 +40,9 @@ const (
 	autoScalingServiceName  = "auto_scaling"
 	cloudwatcherServiceName = "alert"
 	authServiceName         = "auth"
-	kubernetsServiceName    = "kubernetes_engine"
+	kubernetesServiceName   = "kubernetes_engine"
 	containerRegistryName   = "container-registry"
```
Comment (greatbn) [inline client.go:44]: change the name to `container_registry`

Comment (tung491) [reply client.go]: I have renamed it

**Pull request merged**